### DISCUSSION TITLE
Is there a way to check for outdated dependencies, ignoring the semver currently in `.mise.toml`?

### DISCUSSION THREAD

**Discussion by wolverian** [jdx/mise#2659] [Q&A]
I have installed `java@corretto-21.0.3`. Version 21.0.4 is available. Is there a way to make `mise outdated` show me this without loosening the semver in my `.mise.toml` to `corretto-21`? In other words, I would like to use mise to specify the exact version of my dependencies, but also see when new minor releases are available.

**Comment by Marukome0743** [comment]
This is now supported with the `--bump` option, which was added in #2667 and released in mise v2024.9.11.

You can keep the exact version in your configuration and compare it against the latest available version with:

```sh
mise outdated --bump
```

You can also limit the check to Java:

```sh
mise outdated --bump java
```

I verified the behavior on the current version of mise using `java = "corretto-21.0.3"`. A normal `mise outdated java` only compared it with newer releases matching the configured request, while `mise outdated --bump java` reported the current Corretto 21 release (`corretto-21.0.12.8.1` at the time of testing).

The option only broadens the version used for comparison; it does not loosen or rewrite the exact version in your configuration.

*AI-assisted — Tool: Codex; model: OpenAI/GPT-5; version: unavailable.*
